### ISSUE TITLE
Cohort Pathways fails on Impala

### ISSUE THREAD

**Issue by konstjar** [OHDSI/WebAPI#1271]
Odysseus instance, analysis 143
Error:
```
org.ohdsi.webapi.exception.AtlasException: java.lang.RuntimeException: java.util.concurrent.ExecutionException: org.springframework.jdbc.UncategorizedSQLException: StatementCallback; uncategorized SQLException for SQL [update synpuf_53_test_results.i1f0t2a2collapsed_dates_events set cohort_end_date = DATE_ADD(CASE TYPEOF(cohort_end_date ) WHEN 'TIMESTAMP' THEN CAST(cohort_end_date  AS TIMESTAMP) ELSE TO_UTC_TIMESTAMP(CONCAT_WS('-', SUBSTR(CAST(cohort_end_date  AS STRING), 1, 4), SUBSTR(CAST(cohort_end_date  AS STRING), 5, 2), SUBSTR(CAST(cohort_end_date  AS STRING), 7, 2)), 'UTC') END, 1) where cohort_start_date = cohort_end_date]; SQL state [HY000]; error code [500051]; [Simba][ImpalaJDBCDriver](500051) ERROR processing query/statement. Error Code: 0, SQL state: TStatus(statusCode:ERROR_STATUS, sqlState:HY000, errorMessage:AnalysisException: Impala does not support modifying a non-Kudu table: synpuf_53_test_results.i1f0t2a2collapsed_dates_events
), Query: update synpuf_53_test_results.i1f0t2a2collapsed_dates_events set cohort_end_date = DATE_ADD(CASE TYPEOF(cohort_end_date ) WHEN 'TIMESTAMP' THEN CAST(cohort_end_date  AS TIMESTAMP) ELSE TO_UTC_TIMESTAMP(CONCAT_WS('-', SUBSTR(CAST(cohort_end_date  AS STRING), 1, 4), SUBSTR(CAST(cohort_end_date  AS STRING), 5, 2), SUBSTR(CAST(cohort_end_date  AS STRING), 7, 2)), 'UTC') END, 1) where cohort_start_date = cohort_end_date.; nested exception is java.sql.SQLException: [Simba][ImpalaJDBCDriver](500051) ERROR processing query/statement. Error Code: 0, SQL state: TStatus(statusCode:ERROR_STATUS, sqlState:HY000, errorMessage:AnalysisException: Impala does not support modifying a non-Kudu table: synpuf_53_test_results.i1f0t2a2collapsed_dates_events
), Query: update synpuf_53_test_results.i1f0t2a2collapsed_dates_events set cohort_end_date = DATE_ADD(CASE TYPEOF(cohort_end_date ) WHEN 'TIMESTAMP' THEN CAST(cohort_end_date  AS TIMESTAMP) ELSE TO_UTC_TIMESTAMP(CONCAT_WS('-', SUBSTR(CAST(cohort_end_date  AS STRING), 1, 4), SUBSTR(CAST(cohort_end_date  AS STRING), 5, 2), SUBSTR(CAST(cohort_end_date  AS STRING), 7, 2)), 'UTC') END, 1) where cohort_start_date = cohort_end_date.
	at org.ohdsi.webapi.common.generation.TransactionalTasklet.execute(TransactionalTasklet.java:42)
	at org.springframework.batch.core.step.tasklet.TaskletStep$ChunkTransactionCallback.doInTransaction(TaskletStep.java:406)
	at org.springframework.batch.core.step.tasklet.TaskletStep$ChunkTransactionCallback.doInTransaction(TaskletStep.java:330)
	at org.sp
```

**Comment by wivern** [comment]
As mentioned the error is the following:

> Impala does not support modifying a non-Kudu table

I hope used Impala is 5.10 or higher [UPDATE STATEMENT](https://www.cloudera.com/documentation/enterprise/5-10-x/topics/impala_update.html)

And finally [Apache Kudu](https://kudu.apache.org/) storage has to be used for results schema - https://stackoverflow.com/questions/56183013/turn-non-kudu-to-kudu-table-in-impala

**Comment by anthonysena** [comment]
This was closed with the 2.7.3 release.
